EAHW-2813 - Repository Healthcheck (#102)
UKHomeOffice/callisto-timecard-restapi: README.md

# Callisto Timecard REST API

Callisto Timecard REST API is a part of Callisto project.

> The best way to run the service is to leverage the LocalDev solution.

Reviewer: ```suggestion
The best way to run the service is to leverage the [callisto-localdev solution](https://github.com/UKHomeOffice/callisto-localdev).
```

## Running as part of the LocalDev environment

> 1. Download LocalDev repository from https://github.com/UKHomeOffice/callisto-localdev and run it locally as described in [Scenario 1](https://github.com/UKHomeOffice/callisto-localdev#scenario-1-running-callisto-without-need-to-edit-code-base-eg-demo-purposes).

Reviewer: ```suggestion
1. Clone the [callisto-localdev](https://github.com/UKHomeOffice/callisto-localdev) repository and run it locally as described in [Scenario 1](https://github.com/UKHomeOffice/callisto-localdev#scenario-1-running-callisto-without-need-to-edit-code-base-eg-demo-purposes).
```

> 2. From the LocalDev project root, stop the service by running `docker compose stop <service_name>` command.

Reviewer: ```suggestion
2. From the LocalDev project root, stop the service by running `docker compose stop callisto-timecard-restapi` command.
```

3. Ensure you have configured your [Authentication for GitHub Packages](#authentication-for-github-packages)

> 4. Pull the service repository, and from its root directory run command `docker compose up -d`

Reviewer: ```suggestion
4. Clone this repository, and from its root directory run command `docker compose up -d`
```

After successful start, you should be able to work with the service code, and see all changes without restaring the docker container.

## Authentication for Github Packages

In order to pull in Github package dependencies you will need a Github Personal Access Token.
This token will need the minimum of 'packages:read' permissions.

> Update your .m2/settings.xml file to contain the <servers><server> tags

Reviewer: ```suggestion
This can be configured in your .m2/settings.xml file.

E.g.
```

```xml
<servers>
  <server>
      <id>github-packages</id>

>       <username>DummyUser</username>

Reviewer: ```suggestion
      <username>[Insert GitHub username here]</username>
```

>       <password>${env.GITHUB_TOKEN}</password>

Reviewer: ```suggestion
      <password>{[Insert GitHub Personal Access Token here]}</password>
```

Reviewer: I did want the reader to be encouraged to inject their token insecurely by an environment variable instead of encrypting it.

  </server>
</servers>
```

For more info see:
[https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry)

Then run the following to build the project

```sh

> $ mvn clean install

Reviewer: Did we say we'd remove this bit?

Author: yes. If user know what is Java and Spring, probably also know how to run Java/Spring maven project

```

## Debugging

> Assuming you know how to debug java service, the principal is the same when using LocalDev solution.When running using docker compose, the debugger is exposed on the default port (5005). In the case of the debugger port conflicts, change both the ports and the entrypoint section in the docker-compose.yml

Reviewer: ```suggestion
Assuming you know how to debug java service, the principal is the same when using [callisto-localdev solution](https://github.com/UKHomeOffice/callisto-localdev). When running using docker compose, the debugger is exposed on the default port (5005). In the case of the debugger port conflicts, change both the ports and the entrypoint section in the docker-compose.yml
```
